src/coap_session.c

```diff
@@ -631,33 +632,43 @@ coap_session_t *coap_new_client_session_psk(
   if (!session)
     return NULL;
 
-  if (identity) {
+  if (identity && strlen(identity) > 0) {
```

Comment:
Please enclose the comparison in parentheses as always.

Reply:
Fixed.  Brain is not always fully in gear when different projects have different coding styles!